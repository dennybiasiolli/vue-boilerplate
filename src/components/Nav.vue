<template>
  <v-navigation-drawer
    :value="drawer"
    fixed
    app
    @input="toggleDrawer">
    <v-toolbar flat>
      <v-list>
        <v-list-tile>
          <v-list-tile-title class="title">
            Application
          </v-list-tile-title>
        </v-list-tile>
      </v-list>
    </v-toolbar>
    <v-divider />
    <v-list>
      <v-list-tile @click="$router.push('/')">
        <v-list-tile-action>
          <v-icon>dashboard</v-icon>
        </v-list-tile-action>
        <v-list-tile-content>
          <v-list-tile-title>Home</v-list-tile-title>
        </v-list-tile-content>
      </v-list-tile>
      <v-list-tile
        v-for="(c, i) in components"
        :key="i"
        @click="$router.push(c.url)">
        <v-list-tile-action>
          <v-icon>{{ c.icon }}</v-icon>
        </v-list-tile-action>
        <v-list-tile-content>
          <v-list-tile-title>{{ c.title }}</v-list-tile-title>
        </v-list-tile-content>
      </v-list-tile>
    </v-list>
  </v-navigation-drawer>
</template>

<script>
import { createNamespacedHelpers } from 'vuex'

const { mapState, mapMutations } = createNamespacedHelpers('a')

export default {
  name: 'Nav',
  data: () => ({
    components: [
      { title: 'Component 1', url: '/comp1', icon: 'polymer' },
      { title: 'Component 2', url: '/comp2', icon: 'polymer' },
      { title: 'Component 3', url: '/comp3', icon: 'polymer' },
      { title: 'Component 4', url: '/comp4', icon: 'polymer' },
      { title: 'Component 5', url: '/comp5', icon: 'polymer' },
      { title: 'Component 6', url: '/comp6', icon: 'polymer' },
      { title: 'Component 7', url: '/comp7', icon: 'polymer' },
      { title: 'Component 8', url: '/comp8', icon: 'polymer' },
      { title: 'Dashboard', url: '/dashboard', icon: 'dashboard' }
    ]
  }),
  computed: Object.assign({}, mapState(['drawer'])),
  methods: Object.assign({}, mapMutations(['toggleDrawer']))
}
</script>
