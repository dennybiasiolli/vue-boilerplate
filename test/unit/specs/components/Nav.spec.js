import Vuex from 'vuex'
import { shallow, createLocalVue } from '@vue/test-utils'
import Nav from '@/components/Nav'

const localVue = createLocalVue()
localVue.use(Vuex)

const state = {
  drawer: null
}

// Mock the `increment` mutation to make it
// possible to check if it was called.
let mutations

// This function creates a new Vuex store
// instance for every new test case.
const createStore = () => {
  mutations = {
    toggleDrawer: jest.fn()
  }
  return new Vuex.Store({
    modules: {
      a: {
        namespaced: true,
        state,
        mutations
      }
    }
  })
}

describe('Nav', () => {
  test('should match snapshot', () => {
    const component = shallow(Nav, {
      localVue,
      store: createStore()
    })
    expect(component.element).toMatchSnapshot()
  })

  // Evaluate the results of functions in
  // the raw component options
  test('sets the correct components data', () => {
    expect(Nav.name).toBe('Nav')
    expect(typeof Nav.data).toBe('function')
    const defaultData = Nav.data()
    expect(defaultData.components).toEqual([
      { title: 'Component 1', url: '/comp1', icon: 'polymer' },
      { title: 'Component 2', url: '/comp2', icon: 'polymer' },
      { title: 'Component 3', url: '/comp3', icon: 'polymer' },
      { title: 'Component 4', url: '/comp4', icon: 'polymer' },
      { title: 'Component 5', url: '/comp5', icon: 'polymer' },
      { title: 'Component 6', url: '/comp6', icon: 'polymer' },
      { title: 'Component 7', url: '/comp7', icon: 'polymer' },
      { title: 'Component 8', url: '/comp8', icon: 'polymer' },
      { title: 'Dashboard', url: '/dashboard', icon: 'dashboard' }
    ])
    expect(typeof Nav.computed.drawer).toBe('function')
    expect(typeof Nav.methods.toggleDrawer).toBe('function')
  })
})
