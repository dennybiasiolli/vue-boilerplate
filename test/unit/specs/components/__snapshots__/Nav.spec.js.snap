// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match snapshot 1`] = `
<aside
  class="v-navigation-drawer v-navigation-drawer--close v-navigation-drawer--fixed v-navigation-drawer--is-mobile theme--light"
  style="height: 100%; margin-top: 0px; max-height: calc(100% - 0px); width: 300px;"
>
  <nav
    class="v-toolbar elevation-0 theme--light"
    style="margin-top: 0px; padding-right: 0px; padding-left: 0px;"
  >
    <div
      class="v-toolbar__content"
      style="height: 64px;"
    >
      <div
        class="v-list theme--light"
        role="list"
      >
        <div
          role="listitem"
        >
          <div
            class="v-list__tile theme--light"
          >
            <div
              class="v-list__tile__title title"
            >
              
          Application
        
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
   
  <hr
    class="v-divider theme--light"
  />
   
  <div
    class="v-list theme--light"
    role="list"
  >
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__action"
        >
          <i
            aria-hidden="true"
            class="v-icon material-icons theme--light"
          >
            dashboard
          </i>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title"
          >
            Home
          </div>
        </div>
      </a>
    </div>
     
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__action"
        >
          <i
            aria-hidden="true"
            class="v-icon material-icons theme--light"
          >
            polymer
          </i>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title"
          >
            Component 1
          </div>
        </div>
      </a>
    </div>
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__action"
        >
          <i
            aria-hidden="true"
            class="v-icon material-icons theme--light"
          >
            polymer
          </i>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title"
          >
            Component 2
          </div>
        </div>
      </a>
    </div>
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__action"
        >
          <i
            aria-hidden="true"
            class="v-icon material-icons theme--light"
          >
            polymer
          </i>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title"
          >
            Component 3
          </div>
        </div>
      </a>
    </div>
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__action"
        >
          <i
            aria-hidden="true"
            class="v-icon material-icons theme--light"
          >
            polymer
          </i>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title"
          >
            Component 4
          </div>
        </div>
      </a>
    </div>
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__action"
        >
          <i
            aria-hidden="true"
            class="v-icon material-icons theme--light"
          >
            polymer
          </i>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title"
          >
            Component 5
          </div>
        </div>
      </a>
    </div>
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__action"
        >
          <i
            aria-hidden="true"
            class="v-icon material-icons theme--light"
          >
            polymer
          </i>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title"
          >
            Component 6
          </div>
        </div>
      </a>
    </div>
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__action"
        >
          <i
            aria-hidden="true"
            class="v-icon material-icons theme--light"
          >
            polymer
          </i>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title"
          >
            Component 7
          </div>
        </div>
      </a>
    </div>
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__action"
        >
          <i
            aria-hidden="true"
            class="v-icon material-icons theme--light"
          >
            polymer
          </i>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title"
          >
            Component 8
          </div>
        </div>
      </a>
    </div>
    <div
      role="listitem"
    >
      <a
        class="v-list__tile v-list__tile--link theme--light"
      >
        <div
          class="v-list__tile__action"
        >
          <i
            aria-hidden="true"
            class="v-icon material-icons theme--light"
          >
            dashboard
          </i>
        </div>
         
        <div
          class="v-list__tile__content"
        >
          <div
            class="v-list__tile__title"
          >
            Dashboard
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    class="v-navigation-drawer__border"
  />
</aside>
`;
